### PR TITLE
Issue-5791 `gui.animate` doesn't set value if duration = 0 and delay > 0

### LINKED ISSUE #5791: gui.animate doesn't set value if duration = 0 and delay > 0
**Describe the bug (REQUIRED)**
`gui.animate` doesn't set target value if duration = 0 and delay > 0.

**To Reproduce (REQUIRED)**
I attached a re-pro project.

**Expected behavior (REQUIRED)**
`gui.animate` should behave as `go.animate`.

**Defold version (REQUIRED):**
  - Version 1.2.181/1.2.182

**Platforms (REQUIRED):**
 - Platforms: Any

[gui_animate_bug1.zip](https://github.com/defold/defold/files/6420576/gui_animate_bug1.zip)

### PR DESCRIPTION
**Pull request by AGulev**
Fix https://github.com/defold/defold/issues/5791

### CODE REVIEW

File: engine/gui/src/gui.cpp [hunk 1]
```diff
@@ -1716,13 +1716,10 @@ Result DeleteDynamicTexture(HScene scene, const dmhash_t texture_hash)
         for (uint32_t i = 0; i < n; ++i)
         {
             Animation* anim = &(*animations)[i];
-
-            if (anim->m_Elapsed >= anim->m_Duration || anim->m_Cancelled)
+            if ((anim->m_Elapsed >= anim->m_Duration && anim->m_Delay == 0) || anim->m_Cancelled)
```
Comment (JCash) [inline engine/gui/src/gui.cpp:1719]: Perhaps add a unit test for this?

**Review by JCash — DISMISSED** [review]

File: engine/gui/src/gui.cpp [hunk 2]
```diff
@@ -1716,13 +1716,10 @@ Result DeleteDynamicTexture(HScene scene, const dmhash_t texture_hash)
...
             {
                 // If we have cancelled an animation, its callback won't be called which means
                 // we potentially get dangling lua refs in the script system
-                // Another scenario where the callback won't get invoked is when the duration
-                // is 0.0 and the delay > 0.0
```
Comment (AGulev) [inline engine/gui/src/gui.cpp:1725]: this comment is wrong. For now if duration is 0 and delay > 0 we call callback immediately and just ignore delay. Which is wrong and different from what we have in go.animate.

File: engine/gui/src/test/test_gui.cpp [hunk 3]
```diff
@@ -1817,33 +1817,48 @@ TEST_F(dmGuiTest, ScriptAnimate)
...
+        dmGui::SetNodeId(m_Scene, node, "n");
+        char buffer[1024];
 
-    dmGui::Result r;
-    r = dmGui::SetScript(m_Script, LuaSourceFromStr(s));
-    ASSERT_EQ(dmGui::RESULT_OK, r);
-
-    r = dmGui::InitScene(m_Scene);
-    ASSERT_EQ(dmGui::RESULT_OK, r);
+        sprintf(buffer, s, durations[i], delays[i]);
```
Comment (JCash) [inline engine/gui/src/test/test_gui.cpp:1838]: We use `dmSNPrintf(buffer, sizeof(buffer), s, durations[i], delays[i])` from "slib/dstrings.h"

File: engine/gui/src/test/test_gui.cpp [hunk 4]
```diff
@@ -1817,33 +1817,48 @@ TEST_F(dmGuiTest, ScriptAnimate)
...
+                    "function init(self)\n"
+                    "    self.foobar = 123\n"
                     "    self.node = gui.get_node(\"n\")\n"
-                    "    gui.animate(self.node, gui.PROP_POSITION, vmath.vector4(1,0,0,0), gui.EASING_NONE, 1, 0, nil, gui.PLAYBACK_ONCE_BACKWARD)\n"
-                    "end\n"
-                    "function final(self)\n"
-                    "    gui.delete_node(self.node)\n"
+                    "    gui.animate(self.node, gui.PROP_POSITION, vmath.vector4(1,0,0,0), gui.EASING_NONE, %d, %d, cb, gui.PLAYBACK_ONCE_BACKWARD)\n"
                     "end\n";
+    for (int i = 0; i < 9; ++i)
```
Comment (JCash) [inline engine/gui/src/test/test_gui.cpp:1832]: We have `DM_ARRAY_SIZE(durations)` from "dlib/array.h" for things like this.

**Review by JCash — DISMISSED** [review]

**Pull request merged**